js/private/node_package.bzl

```diff
@@ -78,7 +94,52 @@ def _impl(ctx):
     virtual_store_out = None
     node_modules_directory = None
     direct_files = []
-    dep_refs = []
+    direct_dep_refs = []
+
+    if ctx.file.src or ctx.attr.always_output_bins:
+        # output bins for this package
+        for (bin_name, bin_path) in ctx.attr.bins.items():
+            # output a bin entry point this bin
+            bin_out = ctx.actions.declare_file(
+                "{root_dir}/{virtual_store_root}/{virtual_store_name}/node_modules/.bin/{bin_name}".format(
```

Comment:
nit: might be more readable as a `"/".join([ ... ])`

Reply:
Hmm. I actually like the use of `format` here as it separates the variable names from the path segments descriptors. Tho a "/" would be more efficient

Reply:
paths.join with a comment on the structure will work